crates/ethereum/evm/src/execute.rs

```diff
@@ -384,6 +390,71 @@ where
     }
 }
 
+/// An executor that retains all cache state from execution in its bundle state.
+#[derive(Debug)]
+pub struct BlockAccessListExecutor<EvmConfig, DB> {
```

Comment:
Currently we're reading from node components to get the block_executor_provider out, if we're gonna have another executor, are we ok with adding another method maybe called access_list_executor?

On the other side, here's the aforementioned version of code that changes `fn execute(self, xxx)` to `fn execute(&mut self, xxx)`: https://github.com/paradigmxyz/reth/pull/10736/files#diff-eb5f2c91d277e8bdf71850182110f34c23bcb1d70a675f570572f18af090a1d1R33

Reply:
yes, I think we will add another associated type and method to `BlockExecutorProvider`

Reply:
https://github.com/paradigmxyz/reth/pull/10868 tracking here